ethrpc2/block.go

```diff
@@ -0,0 +1,145 @@
+package ethrpc2
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/0xsequence/ethkit/go-ethereum"
+	"github.com/0xsequence/ethkit/go-ethereum/common"
+	"github.com/0xsequence/ethkit/go-ethereum/common/hexutil"
+	"github.com/0xsequence/ethkit/go-ethereum/core/types"
+)
+
+type rpcTransaction struct {
+	tx *types.Transaction
+	txExtraInfo
+}
+
+type txExtraInfo struct {
+	BlockNumber *string         `json:"blockNumber,omitempty"`
+	BlockHash   *common.Hash    `json:"blockHash,omitempty"`
+	From        *common.Address `json:"from,omitempty"`
+	TxType      string          `json:"type,omitempty"`
+}
+
+type rpcBlock struct {
+	Hash         common.Hash      `json:"hash"`
+	Transactions []rpcTransaction `json:"transactions"`
+	UncleHashes  []common.Hash    `json:"uncles"`
+}
+
+func intoBlock(raw json.RawMessage, ret **types.Block) error {
```

Comment:
double pointer..? do we really need that or can we smooth the types?

Reply:
they could also just pass `p.Do(ctx, ethrpc.BlockNumber().Into(blockNum))` and it would just work too with `ret *types.Block` ..no?